crypto/hd/hdpath.go

```diff
@@ -100,16 +99,12 @@ func NewParamsFromPath(path string) (*BIP44Params, error) {
 
 func hardenedInt(field string) (uint32, error) {
 	field = strings.TrimSuffix(field, "'")
-	i, err := strconv.Atoi(field)
 
+	i, err := strconv.ParseUint(field, 10, 32)
 	if err != nil {
 		return 0, err
 	}
 
-	if i < 0 {
-		return 0, fmt.Errorf("fields must not be negative. got %d", i)
```

Comment:
The string `"fields must not be negative. got %d"` is very informative. What will it be replaced with?

Reply:
we don't need it as the string is already cast to uint